:x-pack:plugin:yamlRestTest - hundreds of failures (with specific seed)

:x-pack:plugin:yamlRestTest  with the right seed can result in hundreds of failures. It seems that this happens when the `ml` yamlRestTests are executed prior to the `data_stream` yamlRestTests. This is likely a test not cleaning up after itself. (This is in the x-pack rest tests where a single cluster instance runs hundreds of tests in the same instance)

**Build scan**:

https://gradle-enterprise.elastic.co/s/gibqwc6eawv4c

**Repro line**:
```
git checkout dc0c9942de4594b852acfbc0ead73cc1fc0ce34b 
./gradlew :x-pack:plugin:yamlRestTest -Dtests.seed=275789FA3CA9B3DA --info
```
or more specifically:
```
git checkout dc0c9942de4594b852acfbc0ead73cc1fc0ce34b 
./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=data_stream/*}" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=ml/*}"  -Dtests.seed=275789FA3CA9B3DA --info
```
**Reproduces locally?**:
yes!

**Applicable branches**:
Discovered on master.

**Failure history**:
N/A - the task name was recently changed.

**Failure excerpt**:
The first failure:
```
1> [2020-09-09T23:24:26,341][INFO ][o.e.x.t.r.XPackRestIT    ] [test] [p0=data_stream/80_resolve_index_data_streams/Resolve index with hidden and closed indices] after test
2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=data_stream/80_resolve_index_data_streams/Resolve index with hidden and closed indices}" -Dtests.seed=275789FA3CA9B3DA -Dtests.security.manager=true -Dtests.locale=sr-Latn -Dtests.timezone=Asia/Calcutta -Druntime.java=11 -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*
2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=data_stream/80_resolve_index_data_streams/Resolve index with hidden and closed indices}" -Dtests.seed=275789FA3CA9B3DA -Dtests.security.manager=true -Dtests.locale=sr-Latn -Dtests.timezone=Asia/Calcutta -Druntime.java=11 -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*
2> java.lang.AssertionError: Failure at [data_stream/80_resolve_index_data_streams:132]: indices.3.name didn't match expected value:
                indices.3.name: expected String [test_index1] but was String [.ml-annotations-6]
```

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/80_resolve_index_data_streams.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/80_resolve_index_data_streams.yml
@@ -109,12 +109,12 @@ setup:
 ---
 "Resolve index with hidden and closed indices":
   - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/62190"
+      version: " - 7.8.99"
+      reason: "resolve index api only supported in 7.9+"
 
   - do:
       indices.resolve_index:
-        name: '*'
+        name: ['*','-.ml*']
         expand_wildcards: [all]
 
   - match: {indices.0.name: .ds-simple-data-stream1-000001}

